atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -181,52 +197,81 @@ def get_openshift_session(workflow, namespace, conf_file=None):
     return OSBS(osbs_conf, osbs_conf)
 
 
-def get_group_manifests(workflow):
-    return get_config(workflow)['group_manifests']
+def get_group_manifests(workflow, fallback=NO_FALLBACK):
+    return get_value(workflow, 'group_manifests', fallback)
+
+
+def get_platform_descriptors(workflow, fallback=NO_FALLBACK):
+    return get_value(workflow, 'platform_descriptors', fallback)
+
 
+def get_prefer_schema1_digest(workflow, fallback=NO_FALLBACK):
+    return get_value(workflow, 'prefer_schema1_digest', fallback)
 
-def get_platform_descriptors(workflow):
-    return get_config(workflow)['platform_descriptors']
 
+def get_content_versions(workflow, fallback=NO_FALLBACK):
+    return get_value(workflow, 'content_versions', fallback)
 
-def get_prefer_schema1_digest(workflow):
-    return get_config(workflow)['prefer_schema1_digest']
 
+def get_registries(workflow, fallback=NO_FALLBACK):
+    try:
+        all_registries = get_config(workflow).conf['registries']
+    except KeyError:
+        if fallback != NO_FALLBACK:
+            return fallback
+        raise
 
-def get_content_versions(workflow):
-    return get_config(workflow)['content_versions']
+    registries_cm = {}
+    for registry in all_registries:
+        reguri = RegistryURI(registry.get('url'))
+        regdict = {}
+        regdict['version'] = reguri.version
+        if registry.get('auth'):
+            regdict['secret'] = registry['auth']['cfg_path']
+        regdict['insecure'] = registry.get('insecure', False)
 
+        registries_cm[reguri.docker_uri] = regdict
 
-def get_registries(workflow):
-    return get_config(workflow)['registries']
+    return registries_cm
 
 
-def get_yum_proxy(workflow):
-    return get_config(workflow)['yum_proxy']
+def get_yum_proxy(workflow, fallback=NO_FALLBACK):
+    return get_value(workflow, 'yum_proxy', fallback)
 
 
-def get_source_registry(workflow):
-    return get_config(workflow)['source_registry']
+def get_source_registry(workflow, fallback=NO_FALLBACK):
+    try:
+        source_registry = get_config(workflow).conf['source_registry']
+    except KeyError:
+        if fallback != NO_FALLBACK:
+            return fallback
+        raise
+
+    uri = RegistryURI(source_registry['url']) if source_registry['url'] else None
```

Comment:
I'd rather an exception to propagate if `source_registry['url']` is set to a falsey value. In other words, let's remove conditional check.

Reply:
ok